ui/apps/everest/src/pages/database-form/database-preview/sections/resources-section.tsx

```diff
@@ -44,42 +44,32 @@ export const ResourcesPreviewSection = ({
   const parsedMemory = Number(memory) * intNumberOfNodes;
   const parsedProxyCPU = Number(proxyCpu) * intNumberOfProxies;
   const parsedProxyMemory = Number(proxyMemory) * intNumberOfProxies;
+  const nodesTotalNumber =
+    sharding && shardNr ? +shardNr * intNumberOfNodes : intNumberOfNodes;
+  const nodesText = `${nodesTotalNumber} ${nodesTotalNumber === 1 ? 'node' : 'nodes'}`;
+  const nodesCPUText = `CPU - ${Number.isNaN(parsedCPU) ? '' : `${sharding && shardNr ? (+shardNr * parsedCPU).toFixed(2) : parsedCPU.toFixed(2)} CPU`}`;
+  const nodesMemoryText = `Memory - ${Number.isNaN(parsedMemory) ? '' : `${sharding && shardNr ? (+shardNr * parsedMemory).toFixed(2) : parsedMemory.toFixed(2)} GB`}`;
+  const nodesDiskText = `Disk - ${Number.isNaN(parsedDisk) ? '' : `${sharding && shardNr ? (+shardNr * parsedDisk).toFixed(2) : parsedDisk.toFixed(2)}`} ${diskUnit}`;
```

Comment:
we use `+shardNr` multiple times. Can we refactor into a single var using `parseInt`?